config/initializers/sidekiq.rb

```diff
@@ -1,7 +1,19 @@
-require "sidekiq/web"
-# Sidekiq sessions don't play well with Redis based sessions and Devise
-# <https://github.com/mperham/sidekiq/wiki/Monitoring#sessions-being-lost>
-Sidekiq::Web.set :sessions, false
+# Monkey patch `stringify_keys` to make rack 2.1.1 compatible with Sidekiq UI Admin panel.
+# Should be removed when 2.1.2 is released.
+# https://github.com/rack/rack/pull/1428
+module Rack
```

Comment:
I know I know monkey patching is horrible but I want to give this a shot to see if it fixes the Sidekiq UI problem. I put the patch in this file bc it is solely to fix the Sidekiq issue and it will be easy to remember to remove it bc this file is getting touched a lot. I think the chance of it being forgotten is much higher if we randomly stick it in its own file.

Reply:
With refinements monkey-patching can be made a little bit safer (untested code, but you get the idea):

```ruby
module SidekiqFix
  refine Rack::Session::Abstract::SessionHash do
    def stringify_keys(other)
      other.to_hash.transform_keys(&:to_s)
    end
  end
end

module Rack::Session::Abstract
  using SidekiqFix if/unless # e.g. check Sidekiq version number
end
```
Probably not worth it though.

Reply:
Going to stick with this bc I think its a short term fix but I am definitely going to keep that in mind if we need to monkey patch things in the future more long term. I never thought of doing it that way. Thanks!